app/controllers/puzzles_controller.rb

```diff
@@ -3,14 +3,18 @@ class PuzzlesController < ApplicationController
   before_action :find_puzzle, only: [:show, :edit, :update, :destroy]
 
   def index
-    @material = Material.find_by(id: params.dig(:q, :material_id))
-    @producer = Company.find_by(id: params.dig(:q, :company_id))
-    @inventor = Inventor.find_by(id: params.dig(:q, :inventor_id))
+    @material = Material.find_by(id: params.dig(:ql, :material_id))
+    @producer = Company.find_by(id: params.dig(:ql, :company_id))
+    @inventor = Inventor.find_by(id: params.dig(:ql, :inventor_id))
     @puzzles = @material.puzzles.includes(:rates).order(:name) if @material
     @puzzles = @producer.puzzles.includes(:rates).order(:name) if @producer
     @puzzles = @inventor.puzzles.includes(:rates).order(:name) if @inventor
     @pre_set = @material || @producer || @inventor
-    @puzzles = Puzzle.all.includes(:rates).order(:name) unless @pre_set
+    if !@pre_set
+      @puzzles = Puzzle.search((params.dig(:puzzle, :q) || '*'),
+        fields: [:name, :inventor, :producer], misspellings: { below: 5 },
+         match: :word_middle, order: :name)
```

Comment:
Jeżeli tak chcesz to możesz oczywiście ustawić order, ale weź pod uwagę że domyślnie będziesz miał posortowane po tym jak dobrze się matchuje query - co zazwyczaj ma sens ;)